### DISCUSSION TITLE
How to use script code to open a new graph file in the currently opened nxt_editor without using the File-Open Graph menu of nxt_editor?

### DISCUSSION THREAD

**Discussion by ordinc** [nxt-dev/nxt_editor#261] [Q&A] [answered]
As the title says, I want to use script code to open an already stored .nxt file again in the current editor.
Thanks in advance!

**Comment by ImLucasBrown** [comment]
Hi @ordinc!
What you're trying to do can not be done from within a graph itself. By design, the internal system for compositing and executing graphs has no knowledge of the editor.
However, there is a Python interactive console hidden in the output window. That console has access to the main window instance, which has a method called `load_file`. Here's a screenshot of where that's located and a snipit of code to load a graph.

`main_window.load_file("D:/Projects/nxt/build/release.nxt")`

![image](https://user-images.githubusercontent.com/54835354/234349601-6c620a4d-4046-493b-9292-0ffe7d27f87d.png)

**Reply by ordinc** [reply]
@ImLucasBrown Thanks a lot for the tip.
But how can I get python interactive prompt like your screenshot? I don't have the prompt shown in your screenshot?
![nxt_main](https://user-images.githubusercontent.com/35759494/234501245-d311f871-ad9d-43bb-beea-baac880523fb.gif)

**Reply by ImLucasBrown** [reply]
The text field is invisible, but if you click to the right of the `Python:` label you'll be able to type. In your gif you can see the cursor change to the I-beam when you hover near the bottom.

**Comment by MichaelAldrich** [comment]
I would be willing to guess there's something you can do based on the fact that it's a Qt object with an object name, https://github.com/nxt-dev/nxt_editor/blob/release/nxt_editor/main_window.py#L106

Unfortunatley I don't have an example just ready to go for how to do that, but I'm pretty sure you can use qt to search for children from any widget, like your dcc's main window. Hopefully in the future I'll come back with an example graph.

**Comment by ordinc** [comment]
@ImLucasBrown @MichaelAldrich 
Thanks a lot for your time and tips.
I did some digging in the source code and got this far so far, see gif image below.
![nxt_instance](https://user-images.githubusercontent.com/35759494/234498138-72198ca3-8430-4c2f-a847-6fed80839a8d.gif)

I guess it should be that the new instance needs to get the object of the editor that I have opened so that I can achieve my goal and open it in the current editor instead of opening a new editor UI again.
I also communicated with our technical art, if I succeed, I will come back here.
thank you all.

**Reply by ImLucasBrown** [reply, marked as answer]
Unfortunately this doesn't work as expected because the NXT main window isn't a singleton. Here's some code I've used in the past to do something similar to what I think you're trying to do here.

```python
from nxt_editor.main_window import MainWindow
from Qt import QtWidgets

qapp = QtWidgets.QApplication.instance()
if not qapp:
    raise Exception('No qapp found! Is nxt open?')
windows = qapp.topLevelWidgets()
for win in windows:
    if isinstance(win, MainWindow):
        nxt_window = win
        break
else:
    raise Exception('No nxt window found')
```

Qt knows about all the top level widgets, so if we loop over them looking for one that's an instance of the NXT MainWindow you should be able to find it. Keep in mind this code will only return the first NXT instance it finds, if you have multiple open it may return one you weren't trying to interact with.

**Reply by ordinc** [reply]
Great, this one does the job of the title as expected.
Thanks for the tip.
